Allow setting all cookie package serialize/parse options (#9063)
withastro/astro: .changeset/tiny-days-dance.md

---
'astro': minor
---

> Cookie encoding / decoding can now be customized

Reviewer: What's the convention for ending periods @sarah11918 ? Curious for future changesets and consistency

Reviewer 2: The first line gets used as a heading in the blog post, so that's why I always omit the period there.

Reviewer: Gotcha, thanks!

> New `encode` and `decode` functions allow customizing how cookies are encoded and decoded. For example, cookies are by default encoded via `encodeURIComponent`, but you might want to avoid that in some cases such as when adding a URL as part of a cookie.

Reviewer 3: ```suggestion
Adds new `encode` and `decode` functions to allow customizing how cookies are encoded and decoded. For example, you can bypass the default encoding via `encodeURIComponent` when adding a URL as part of a cookie:
```

```astro

> ---
> Astro.cookies.set('url', Astro.url.toString(), {
>   // Override the default encoding so that URI components are not encoded
>   encode: o => o
> });
> ---

Reviewer 4: ```suggestion
---
import { encodeCookieValue } from "./cookies";
Astro.cookies.set('url', Astro.url.toString(), {
  // Override the default encoding so that URI components are not encoded
  encode: value => encodeCookieValue(value)
});
---
```

Reviewer 3: I like Ema's changes to the code sample!

Reviewer 2: Yes! Thanks so much, i was struggling with this, as this is *really* the use-case in mind, so just giving it a function name does help a lot.

```

> Later, you can get back the URL in the same way:

Reviewer 3: ```suggestion
Later, you can decode the URL in the same way:
```

```astro

> ---
> const url = Astro.cookies.get('url', {
>   decode: o => o
> });
> ---

Reviewer 4: ```suggestion
---
import { decodeCookieValue } from "./cookies";
const url = Astro.cookies.get('url', {
  decode: value => decodeCookieValue(value)
});
---
```

---
'astro': minor
---

> Allows specifying `encode` and `decode` functions for cookies

Reviewer 5: cc @florian-lefebvre

> Allows specifying `encode` and `decode` functions for cookies

Reviewer: Will do that tomorrow

> Allows specifying `encode` and `decode` functions for cookies

Reviewer 2: @florian-lefebvre No need, I already redid the changeset. Feel free to improve mine though!
